### ISSUE TITLE
systemd-backlight@.service - Failed to get backlight or LED device 'backlight:acpi_video0': No such device

### ISSUE THREAD

**Issue by quietpinguin** [systemd/systemd#21997]
**systemd version the issue has been seen with**
 > 250-5

**Used distribution**
 > Manjaro

**Linux kernel version used**
 > `Linux Manjaro 5.15.12-1-MANJARO #1 SMP PREEMPT Wed Dec 29 18:08:07 UTC 2021 x86_64 GNU/Linux`

**CPU architecture issue was seen on**
 > AMD Ryzen 5 3500U with Radeon Vega Mobile Gfx

**Expected behaviour you didn't see**
 > None - laptop display can be correctly adjusted using function keys

**Unexpected behaviour you saw**
 > In addition to correctly restoring backlight brightness setting for display device amdgpu_bl0, systemd-backlight@.service tries to restore backlight brightness setting for **non-existent** display device acpi_video0
 > I would expect systemd-backlight@.service to check for available display devices before trying to restore brightness settings.

**Steps to reproduce the problem**
 > Start or restart laptop - unexpected behaviour occurs on boot before sddm is launched

**Additional program output to the terminal or log subsystem illustrating the issue**

```04.01.22 09:23	systemd	Created slice Slice /system/systemd-backlight.
04.01.22 09:23	systemd	Starting Load/Save Screen Backlight Brightness of backlight:acpi_video0...
04.01.22 09:23	systemd	Starting Load/Save Screen Backlight Brightness of backlight:amdgpu_bl0...
04.01.22 09:23	systemd-backlight	Failed to get backlight or LED device 'backlight:acpi_video0': No such device
04.01.22 09:23	systemd	systemd-backlight@backlight:acpi_video0.service: Main process exited, code=exited, status=1/FAILURE
04.01.22 09:23	systemd	systemd-backlight@backlight:acpi_video0.service: Failed with result 'exit-code'.
04.01.22 09:23	systemd	Failed to start Load/Save Screen Backlight Brightness of backlight:acpi_video0.
04.01.22 09:23	audit	SERVICE_START pid=1 uid=0 auid=4294967295 ses=4294967295 subj==unconfined msg='unit=systemd-backlight@backlight:acpi_video0 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=failed'
04.01.22 09:23	systemd	Finished Load/Save Screen Backlight Brightness of backlight:amdgpu_bl0.
04.01.22 09:23	audit	SERVICE_START pid=1 uid=0 auid=4294967295 ses=4294967295 subj==unconfined msg='unit=systemd-backlight@backlight:amdgpu_bl0 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
04.01.22 09:23	dbus-daemon	[system] Activating service name='org.kde.powerdevil.backlighthelper' requested by ':1.36' (uid=1000 pid=801 comm="/usr/lib/org_kde_powerdevil ") (using servicehelper)'''

**Comment by yuwata** [comment]
Could you add the following udev rules file:
```
# /etc/udev/rules.d/00-debug-backlight.rules 
SUBSYSTEM=="backlight", OPTIONS="log_level=debug"
```
then reboot the system and provide the journal log with
```
journalctl -b --no-hostname -u systemd-backlight@backlight:acpi_video0.service -u systemd-udevd.service
```

**Comment by quietpinguin** [comment]
Herewith the journal log using journalctl -b --no-hostname -u 
***@***.***:acpi_video0.service -u systemd-udevd.service

Jan 05 08:48:29 systemd[1]: Started Rule-based Manager for Device Events 
and Files.
Jan 05 08:48:30 mtp-probe[346]: checking bus 3, device 2: 
"/sys/devices/pci0000:00/0000:00:08.1/0000:03:00.4/usb3/3-1"
Jan 05 08:48:30 mtp-probe[346]: bus: 3, device: 2 was not an MTP device
Jan 05 08:48:30 mtp-probe[413]: checking bus 3, device 5: 
"/sys/devices/pci0000:00/0000:00:08.1/0000:03:00.4/usb3/3-2/3-2.2"
Jan 05 08:48:30 mtp-probe[413]: bus: 3, device: 5 was not an MTP device
Jan 05 08:48:31 mtp-probe[539]: checking bus 3, device 5: 
"/sys/devices/pci0000:00/0000:00:08.1/0000:03:00.4/usb3/3-2/3-2.2"
Jan 05 08:48:31 mtp-probe[539]: bus: 3, device: 5 was not an MTP device
Jan 05 08:48:31 systemd-udevd[328]: acpi_video0: The log level is 
changed to 'debug' while processing device (SEQNUM=2831, ACTION=add)
Jan 05 08:48:31 systemd-udevd[319]: amdgpu_bl0: The log level is changed 
to 'debug' while processing device (SEQNUM=3151, ACTION=add)
Jan 05 08:48:31 systemd-udevd[328]: acpi_video0: 
/usr/lib/udev/rules.d/71-seat.rules:74 Importing properties from results 
of builtin command 'path_id'
Jan 05 08:48:31 systemd-udevd[319]: amdgpu_bl0: 
/usr/lib/udev/rules.d/71-seat.rules:74 Importing properties from results 
of builtin command 'path_id'
Jan 05 08:48:31 systemd-udevd[319]: amdgpu_bl0: 
/usr/lib/udev/rules.d/99-systemd.rules:64 Skipping builtin 'path_id' in 
IMPORT key
Jan 05 08:48:31 systemd-udevd[319]: amdgpu_bl0: sd-device: Created db 
file '/run/udev/data/+backlight:amdgpu_bl0' for 
'/devices/pci0000:00/0000:00:08.1/000>
Jan 05 08:48:31 systemd-udevd[319]: amdgpu_bl0: Device processed 
(SEQNUM=3151, ACTION=add)
Jan 05 08:48:31 systemd-udevd[319]: amdgpu_bl0: sd-device-monitor: 
Passed 385 byte to netlink monitor
Jan 05 08:48:31 systemd-udevd[328]: acpi_video0: 
/usr/lib/udev/rules.d/99-systemd.rules:64 Skipping builtin 'path_id' in 
IMPORT key
Jan 05 08:48:31 systemd-udevd[328]: acpi_video0: sd-device: Created db 
file '/run/udev/data/+backlight:acpi_video0' for 
'/devices/pci0000:00/0000:00:08.1/0>
Jan 05 08:48:31 systemd-udevd[328]: acpi_video0: Device processed 
(SEQNUM=2831, ACTION=add)
Jan 05 08:48:31 systemd-udevd[328]: acpi_video0: sd-device-monitor: 
Passed 387 byte to netlink monitor
Jan 05 08:48:31 systemd-udevd[313]: acpi_video0: The log level is 
changed to 'debug' while processing device (SEQNUM=3152, ACTION=remove)
Jan 05 08:48:31 systemd-udevd[313]: acpi_video0: Device processed 
(SEQNUM=3152, ACTION=remove)
Jan 05 08:48:31 systemd-udevd[313]: acpi_video0: sd-device-monitor: 
Passed 390 byte to netlink monitor
Jan 05 08:48:31 systemd[1]: Starting Load/Save Screen Backlight 
Brightness of backlight:acpi_video0...
Jan 05 08:48:31 systemd-backlight[582]: Failed to get backlight or LED 
device 'backlight:acpi_video0': No such device
Jan 05 08:48:31 systemd[1]: 
***@***.***:acpi_video0.service: Main process exited, 
code=exited, status=1/FAILURE
Jan 05 08:48:31 systemd[1]: 
***@***.***:acpi_video0.service: Failed with result 
'exit-code'.
Jan 05 08:48:31 systemd[1]: Failed to start Load/Save Screen Backlight 
Brightness of backlight:acpi_video0.
Jan 05 08:48:31 systemd-udevd[330]: amdgpu_bl0: The log level is changed 
to 'debug' while processing device (SEQNUM=3169, ACTION=change)
Jan 05 08:48:31 systemd-udevd[330]: amdgpu_bl0: 
/usr/lib/udev/rules.d/71-seat.rules:74 Importing properties from results 
of builtin command 'path_id'
Jan 05 08:48:31 systemd-udevd[330]: amdgpu_bl0: 
/usr/lib/udev/rules.d/99-systemd.rules:64 Skipping builtin 'path_id' in 
IMPORT key
Jan 05 08:48:31 systemd-udevd[330]: amdgpu_bl0: sd-device: Created db 
file '/run/udev/data/+backlight:amdgpu_bl0' for 
'/devices/pci0000:00/0000:00:08.1/000>
Jan 05 08:48:31 systemd-udevd[330]: amdgpu_bl0: Device processed 
(SEQNUM=3169, ACTION=change)
Jan 05 08:48:31 systemd-udevd[330]: amdgpu_bl0: sd-device-monitor: 
Passed 401 byte to netlink monitor

**Comment by yuwata** [comment]
@quietpinguin Thanks. Fix is waiting in #22011.

**Comment by yuwata** [comment]
(But, I think the driver for AMD GPU should not remove the acpi device... The backlight handling of AMD GPU driver is always broken...)

**Comment by quietpinguin** [comment]
On my laptop the backlight handling of AMD GPU driver works perfectly, 
even though the acpi device has been removed...

I think Yu Watanabe's approach is correct, if the acpi driver has been 
removed (for what ever reason) then there is no need to publish an error 
message for a backlight setting that is no longer needed.

**Comment by xReav3r** [comment]
Hello,
I have got same problem. On Manjaro 21.2.0/21.2.1 the brightness is set to 0 (black screen, but visible when provided external light) but on previous versions I was at ~50% of brightness fixed.

Managed to solve it by installing pop-os but ran into issues after updates:
https://github.com/pop-os/pop/issues/1976
https://github.com/pop-os/pop/issues/2026

On Manjaro 21.2.1 it fails for me on:
Failed to get backlight or LED device 'backlight:acpi_video0': No such device
Failed to start Load/Save Screen Backlight Brightness of backlight:acpi_video0.

Interesting is that in worked on specific version of kernel.
